Bio/PDB/mmtf/mmtfio.py

```diff
@@ -115,6 +116,22 @@ def _save_structure(self, filepath, select):
             experimental_methods=header_dict["structure_method"],
         )
 
+        chain_ids = [chain.get_id() for chain in self.structure.get_chains()]
+        if "biomoltrans" in header_dict:
+            biomoltrans = header_dict["biomoltrans"]
+            for key, value in biomoltrans.items():
+                matrix_items = []  # list of 16 items of 4x4 matrix
+                for line in value[1:]:
+                    # 3 lines for 3x3 rotation and last column translations
+                    matrix_items.extend([float(item) for item in line.split()])
+                matrix_items.extend([0.0, 0.0, 0.0, 1.0])
+
+                encoder.set_bio_assembly_trans(
+                    bio_assembly_index=key,
+                    input_chain_indices=[chain_ids.index(c) for c in value[0]],
```

Comment:
Small nitpick: for small structures this won't matter but for larger ones (e.g. viral capsids) with hundreds of chains, it might be faster to compile the chain_ids into a dict with enumerate and use it here. `list.index` is O(n).

```python
chain_id_to_idx = {v: k for k, v in enumerate(c.id for c in self.structure.get_chains())}
```

Reply:
Makes sense. Thanks

Reply:
I just spotted that this calculation had already been done outside the loop and put into a variable so I'm changing to using that